lib/pbench/test/unit/server/query_apis/test_datasets_detail.py

```diff
@@ -189,6 +189,152 @@ def test_query(
+
+        response = query_api(
+            "/datasets/detail",
+            "/_search?ignore_unavailable=true",
+            payload,
+            index,
+            HTTPStatus.OK,
+            headers={"authorization": f"Bearer {pbench_token}"},
+            json=response_payload,
+        )
+        assert response.status_code == HTTPStatus.OK
+        res_json = response.json
+
+        # NOTE: we asked for "seen" and "deleted" metadata, but the "deleted"
+        # key wasn't created, so we verify that only "seen" is present in the
+        # response.
+        expected = {
+            "hostTools": [
+                {
+                    "hostname": "dhcp31-187",
+                    "tools": {
+                        "iostat": "--interval=3",
+                        "mpstat": "--interval=3",
+                        "perf": "--record-opts='record -a --freq=100'",
+                        "pidstat": "--interval=30",
+                        "proc-interrupts": "--interval=3",
+                        "proc-vmstat": "--interval=3",
+                        "sar": "--interval=3",
+                        "turbostat": "--interval=3",
+                    },
+                }
+            ],
+            "runMetadata": {
+                "config": "rhel8_kvm_perf43_preallocfull_nvme_run4_iothread_isolcpus",
+                "controller": "node",
+                "controller_dir": "dhcp31-187.example.com",
+                "date": "2020-04-29T12:48:33",
+                "end": "2020-04-29T13:30:04.918704",
+                "file-date": "2020-11-20T21:01:54.532281",
+                "file-name": "/pbench/archive/fs-version-001/dhcp31-187.example.com/fio_rhel8_kvm_perf43_preallocfull_nvme_run4_iothread_isolcpus_2020.04.29T12.49.13.tar.xz",
+                "file-size": 216319392,
+                "id": "12fb1e952fd826727810868c9327254f",
+                "iterations": "0__bs=4k_iodepth=1_iodepth_batch_complete_max=1, 1__bs=32k_iodepth=1_iodepth_batch_complete_max=1, 2__bs=256k_iodepth=1_iodepth_batch_complete_max=1, 3__bs=4k_iodepth=8_iodepth_batch_complete_max=8, 4__bs=32k_iodepth=8_iodepth_batch_complete_max=8, 5__bs=256k_iodepth=8_iodepth_batch_complete_max=8, 6__bs=4k_iodepth=16_iodepth_batch_complete_max=16, 7__bs=32k_iodepth=16_iodepth_batch_complete_max=16, 8__bs=256k_iodepth=16_iodepth_batch_complete_max=16",
+                "md5": "12fb1e952fd826727810868c9327254f",
+                "name": "drb",
+                "pbench-agent-version": "0.68-1gf4c94b4d",
+                "raw_size": 292124533,
+                "script": "fio",
+                "start": "2020-04-29T12:49:13.560620",
+                "tar-ball-creation-timestamp": "2020-04-29T15:16:51.880540",
+                "toc-prefix": "fio_rhel8_kvm_perf43_preallocfull_nvme_run4_iothread_isolcpus_2020.04.29T12.49.13",
+                "toolsgroup": "default",
+            },
+            "serverMetadata": {"seen": True},
+        }
+        assert expected == res_json
```

Comment:
Given that this is driven by a mocked response, do we really need to include _all_ of these fields in the mock response and then have to deal with them in the expected value?  Can't we pare this down a bunch?  That would make the test easier to read and make it easier to see what's actually being tested.

(Ditto for the other tests like this one....)

Reply:
Realism vs simplicity ... and it was easiest to just copy and paste from a real query. (Though I've changed a few of the fields along the way for various reasons.)